src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLSPTextViewConnectionListener.cs

```diff
@@ -233,17 +234,26 @@ private async void RazorOptions_OptionChanged(object sender, EditorOptionChanged
             // options later when the server sends us a workspace/configuration request.
             _clientOptionsMonitor.UpdateOptions(settings);
 
-            // Make sure the server updates the settings on their side by sending a
-            // workspace/didChangeConfiguration request. This notifies the server that the user's
-            // settings have changed. The server will then query the client's options monitor (already
-            // updated via the line above) by sending a workspace/configuration request.
-            // NOTE: This flow uses polyfilling because VS doesn't yet support workspace configuration
-            // updates. Once they do, we can get rid of this extra logic.
-            await _requestInvoker.ReinvokeRequestOnServerAsync<DidChangeConfigurationParams, Unit>(
-                Methods.WorkspaceDidChangeConfigurationName,
-                RazorLSPConstants.RazorLSPContentTypeName,
-                new DidChangeConfigurationParams(),
-                CancellationToken.None);
+            try
+            {
+                // Make sure the server updates the settings on their side by sending a
+                // workspace/didChangeConfiguration request. This notifies the server that the user's
+                // settings have changed. The server will then query the client's options monitor (already
+                // updated via the line above) by sending a workspace/configuration request.
+                // NOTE: This flow uses polyfilling because VS doesn't yet support workspace configuration
+                // updates. Once they do, we can get rid of this extra logic.
+                await _requestInvoker.ReinvokeRequestOnServerAsync<DidChangeConfigurationParams, Unit>(
+                    Methods.WorkspaceDidChangeConfigurationName,
+                    RazorLSPConstants.RazorLSPContentTypeName,
+                    new DidChangeConfigurationParams(),
+                    CancellationToken.None);
+            }
+            catch (Exception ex)
+            {
+                // This shouldn't happen, but VS will crash if an exception is hit in this method so
+                // we should catch any exceptions here just in case.
+                Debug.Fail($"Error executing workspace/didChangeConfiguration request: {ex.Message}");
```

Comment:
So ideally I would have wanted to add some logging here, but I don't know if it's possible since the logging infrastructure doesn't get initialized until later? Any thoughts @NTaylorMullen @TanayParikh?

Reply:
Thought about the same thing. Technically we could force initialize the HtmlC# logger but not sure it's worth it. Ideally the LSP platform should hold all requests until a language server has fully booted

Reply:
We do force initialize in some other cases such as the HTML Diagnostics Interceptor which isn't part of our "normal" language server lifecycle. Here's an example of how it _could_ be done:
https://github.com/dotnet/aspnetcore-tooling/blob/main/src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorHtmlPublishDiagnosticsInterceptor.cs#L153-L160

Regarding whether or not it should be, I agree with Taylor it would be nice if LSP platform queued the requests for us.